libraries/botbuilder-ai/tests/qnaMaker.test.js

```diff
@@ -70,111 +71,298 @@ describe('QnAMaker', function () {
         .replace(/ /g, '_');
     }
 
-    it('should work free standing', async function () {
-        const qna = new ai.QnAMaker(endpoint, { top: 1 });
-        const answer = 'BaseCamp: You can use a damp rag to clean around the Power Pack';
+    describe('Instantiation', function() {
+        it('should instantiate a QnAMaker class successfully with QnAMakerEndpoint, but without options specified', function() {
+            const qnaWithoutOptions = new QnAMaker(endpoint);
+            const isQnaMaker = qnaWithoutOptions instanceof QnAMaker;
 
-        const firstCallQnaResults = await qna.generateAnswer(`how do I clean the stove?`);
-        assert.notStrictEqual(firstCallQnaResults, true, `The response was returned as 'undefined'.`);
-        assert.strictEqual(firstCallQnaResults.length, 1, 'Should have received just one answer on the first call.');
-        assert.strictEqual(firstCallQnaResults[0].answer.startsWith(answer), true, `The answer should have started with '${answer}' for the first call.`);
+            assert.strictEqual(isQnaMaker, true, 'Should create QnAMaker instance successfully with only QnAMakerEndpoint arg.');
+        });
 
-        const secondCallQnaResults = await qna.generateAnswer("is the stove hard to clean?");
-        assert.notStrictEqual(secondCallQnaResults, true, `The response was returned as 'undefined'.`);
-        assert.strictEqual(secondCallQnaResults.length, 1, 'Should have received just one answer on the second call.');
-        assert.strictEqual(secondCallQnaResults[0].answer.startsWith(answer), true, `The answer should have started with '${answer}' for the second call.`);
-    });
+        it('should instantiate a QnAMaker class successfully with both QnAMakerEndpoint and QnAMakerOptions args', function() {
+            const options = { top: 7 };
+            const qnaWithOptions = new QnAMaker(endpoint, options);
+            const isQnaMaker = qnaWithOptions instanceof QnAMaker;
 
-    it('should return 0 answers for a question with no answer after a succesful call', async function () {
-        const qna = new ai.QnAMaker(endpoint, { top: 1 });
-        let answer = 'BaseCamp: You can use a damp rag to clean around the Power Pack';
+            assert.strictEqual(isQnaMaker, true, 'Should create QnAMake instance successfully with QnAMakerOptions.');
```

Comment:
While this wouldn't be available in TypeScript, in JavaScript you can evaluate the value of `qnaWithOptions._options` to verify that the options were successfully passed in. The feedback on the previous test is applicable to here as well.